### PR TITLE
#170764597: Like or unlike accommodation facility

### PR DESCRIPTION
**Pull request by KarangwaEmmy**
### What does this PR do?
Enables a user to like accommodation
Enables a user to unlike accommodation
### Description of Task to be completed?
add like to accommodation
remove like to accommodation
check if accommodation to liked or unliked is available
### How should this be manually tested?
git clone this repo and check out to `ft-like-unlike-accomodation-170764597`
run `npm I` to install all the dependencies
run sequelize db:migrate:undo: all then npm run migrate to add the like schema
run the application with npm run dev
register accommodation in the system

login the user and user the token provided in subsequent operations
- Use Postman: send a POST request to `/accommodations/:id/like`  
### Any background context you want to provide?
N/A
What are the relevant pivotal tracker stories?
#170764597

Screenshots (if appropriate)

like

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -53,13 +53,15 @@
...
   "scripts": {
+    "pre-dev": "cross-env NODE_ENV=test npm run undo:db && cross-env NODE_ENV=test npm run migrate:db && cross-env NODE_ENV=dev npm run seed:db",
     "pre-test": "cross-env NODE_ENV=test npm run undo:db && cross-env NODE_ENV=test npm run migrate:db && cross-env NODE_ENV=test npm run seed:db",
     "test": "npm run pre-test && cross-env NODE_ENV=test nyc --reporter=text --reporter=html mocha --timeout=5000 --require @babel/polyfill --require @babel/register --exit ./src/test/index.test.js",
     "stop-linux": "killall node",
     "start": "cross-env NODE_ENV=production npm run undo:db && cross-env NODE_ENV=production npm run migrate:db && cross-env NODE_ENV=production npm run build && node dist/server.js",
     "build": "rm -rf dist && mkdir dist && babel src -s -d dist --copy-files",
     "coveralls": "nyc report --reporter=text-lcov | coveralls",
     "dev": "node_modules/.bin/nodemon ./src/server.js --exec babel-node --",
+    "devs": "npm run pre-test && cross-env NODE_ENV=dev & nodemon ./src/server.js --exec babel-node --",
```
Comment (shaazk) [inline package.json]: What is this script for? I don't think you need it

File: package.json [hunk 2]
```diff
@@ -53,13 +53,15 @@
     "sequelize-migration-pg-extras": "^1.0.1"
   },
   "scripts": {
+    "pre-dev": "cross-env NODE_ENV=test npm run undo:db && cross-env NODE_ENV=test npm run migrate:db && cross-env NODE_ENV=dev npm run seed:db",
```
Comment (shaazk) [inline package.json]: I don't think you need this script too. Is there any reason you added it?

**Review by shaazk — CHANGES_REQUESTED** [review]

File: src/database/migrations/20200205141154-create-accomodation.js [hunk 3]
```diff
@@ -45,18 +44,16 @@ export default {
...
     createdAt: {
       allowNull: false,
-      type: Sequelize.DATE,
-      defaultValue: Sequelize.fn('NOW')
+      type: Sequelize.DATE
     },
     updatedAt: {
       allowNull: false,
-      type: Sequelize.DATE,
-      defaultValue: Sequelize.fn('NOW')
```
Comment (shaazk) [inline src/database/migrations/20200205141154-create-accomodation.js]: Why did you delete the default value constraint?

**Review by shaazk — CHANGES_REQUESTED** [review]

File: src/database/migrations/20200205141154-create-accomodation.js (hunk 3 above)
Comment (KarangwaEmmy) [reply src/database/migrations/20200205141154-create-accomodation.js]: I have fixed issue.

**Review by shaazk — APPROVED** [review]

**Review by Elvis-rugamba — APPROVED** [review]

File: src/controllers/LikeController.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        return response.sendErrorMessage();
+      }
+      Likes = await LikeService.accommodationLikes(AccommodationId);
+      alreadyLiked = await LikeService.CountLikes({
+        user_id: userData.id,
+        accommodation_id: AccommodationId
+      });
+      if (alreadyLiked) {
+        await LikeService.unLike({ user_id: userData.id, accommodation_id: AccommodationId });
+        return res.status(201).json({ status: 201, message: `YOu have unliked ${accommodation.name}`, likes: Likes - 1 });
```
Comment (ajimae) [inline src/controllers/LikeController.js]: Correct the spelling `Y0u`

File: src/controllers/LikeController.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+      }
+      const addedLike = await LikeService.Like({
+        user_id: userData.id,
+        accommodation_id: AccommodationId,
+      }, {
+        fields: ['user_id', 'accommodation_id']
+      });
+
+      if (addedLike) {
+        return res.status(201).json({ status: 201, message: `YOu have liked ${accommodation.name}`, likes: Likes + 1 });
```
Comment (ajimae) [inline src/controllers/LikeController.js]: Same here, correct `Y0u`

File: src/repositories/LikeRepository.js [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+class LikeRepository {
+  /**
+   * Creates a new Like.
+   * @param {object} user_id
+   * @param {object} accommodation_id
+   * @returns {object} Like.
+   */
+  static async Like({ user_id, accommodation_id }) {
+    try {
+      return await Like.create({
```
Comment (ajimae) [inline src/repositories/LikeRepository.js]: When you have a `return` you don't need the await anymore, this is redundant. Try removing the `await` or assign it to a variable and return the variable instead.

File: src/repositories/LikeRepository.js [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+  }
+
+  /**
+   * check if accommodation exists.
+   * @param {object} accommodation_id .
+   * @returns {object} Like object.
+   */
+  static async checkAccommodation(accommodation_id) {
+    try {
+      return await Accommodation.findOne({
```
Comment (ajimae) [inline src/repositories/LikeRepository.js]: Same here, remove the `await` keyword or assign to a variable that you'll be returning.

File: src/repositories/LikeRepository.js [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+
+  /**
+   * Check if like exists by user id and accommodatio id.
+   * @param {object} user_id.
+   * @param {object} accommodation_id.
+   * @returns {object} Like object.
+   */
+  static async checkLikes({ user_id, accommodation_id }) {
+    try {
+      return await Like.findOne({
```
Comment (ajimae) [inline src/repositories/LikeRepository.js]: Same here

File: src/controllers/LikeController.js (hunk 4 above)
Comment (ajimae) [inline src/controllers/LikeController.js]: I believe you guys have a utility function e.g `const response = new Response(res, 404, 'accommodation not found');` that does this, why are we still doing this?

**Review by ajimae — COMMENTED** [review]
@KarangwaEmmy I left some comment on your PR please endeavour to look at them and apply the necessary changes. Thanks.

**Review by ajimae — APPROVED** [review]
LGTM!
